modules/6_clusterqc/rna/clusterqc.qmd

```diff
@@ -0,0 +1,309 @@
+
+# Set default assay
+default_assay = param("default_assay")
+Seurat::DefaultAssay(sc) = default_assay
+```
+
+```{r}
+#| label: temporary
+#| warning: false
+#| message: false
+sc <- suppressMessages(ScaleData(sc))
+sc <- suppressMessages(RunPCA(sc))
+sc <- suppressMessages(FindNeighbors(sc, dims = 1:10))
+sc <- suppressMessages(FindClusters(sc, resolution = 0.5))
+sc <- suppressMessages(RunUMAP(sc, dims = 1:10))
+```
+## Quality Control 
+
+Do cells in individual clusters have particularly high counts, detected genes or mitochondrial content?
+
+```{r}
+#| label: fig-clusterqc
+#| warning: false
+#| message: false
+#| results: asis
+
+qc_features = paste(c("nCount", "nFeature", "pMito", "pRibosomal", "pGlobin", "pERCC", "pXIST", "pChrY"), default_assay, sep="_")
+if ("DoubletScore" %in% colnames(sc[[]])) qc_features = c(qc_features, "DoubletScore")
+
+# Feature plots
+plist1 = Seurat::FeaturePlot(sc, features=qc_features, combine=FALSE)
+for (i in seq(plist1)) {
+  plist1[[i]] = plist1[[i]] + AddPlotStyle(title="") + viridis::scale_color_viridis()
+}
+names(plist1) = qc_features
+if (default_assay == "RNA") plist1[["nCount_RNA"]] = plist1[["nCount_RNA"]] + viridis::scale_color_viridis(trans="log10")
+
+# Violin plots
+plist2 = lapply(qc_features, function(x) {
+  ggplot(sc[[]], aes(x=seurat_clusters, y=!!sym(x), fill=seurat_clusters, group=seurat_clusters)) + 
+  geom_violin(scale="width") + 
+  AddPlotStyle(fill=ScColours(sc, "seurat_clusters"), xlab="Cluster", legend_position="none")
+})
+names(plist2) = qc_features
+
+plist = lapply(qc_features, function(x) {
+  return(plist1[[x]] + plist2[[x]])
+})
+captions = GeneratePlotCaptions(qc_features, remove=paste0("_", default_assay), capitalize=TRUE)
+names(plist) = names(captions) = qc_features
+
+# Set up layout and generate chunks
+chunk_template = "
+##### {{type}}
+
+\`\`\`{r}
+#| label: fig-clusterqc_rna_{{type}}
+#| fig-cap: '{{caption}}'
+#| fig-height: 4
+
+print(plist[['{{type}}']])
+\`\`\`
+"
+
+cat("::: panel-tabset", sep="\n")
+
+for (x in qc_features) {
+  chunk_filled =  knitr::knit_expand(text=chunk_template, type=x, caption=captions[x])
+  if(interactive()) {
+    EvalKnitrChunk(chunk_filled)
+  } else {
+    chunk_filled = knitr::knit_child(text=chunk_filled, envir=environment(), quiet=TRUE)
+    cat(chunk_filled, sep='\n')
+  }
+}
+
+cat(":::", sep="\n")
+```
+
+## Cell Cycle Effects
+
+How much do gene expression profiles in the dataset reflect the cell cycle phases the single cells were in? After initial normalisation, we determined the effects of cell cycle heterogeneity by calculating a score for each cell based on its expression of G2M and S phase markers. Scoring is based on the strategy described in [@Tirosh_2016], and human gene symbols are translated to gene symbols of the species of interest using biomaRt. This section of the report visualises the above calculated cell cycle scores. 
+
+```{r}
+#| label: fig-cellcycleeffects
```

Comment:
Can we stay consistent with the naming? How about `fig-clusterqc_rna_cellcycle` ?

Reply:
Moreover should chunk labels use minus or underscores?